### ISSUE TITLE
"Event index" should no longer be used to identify the row# or image# in an .rtdc file 

### ISSUE THREAD

**Issue by chrherold** [ZELLMECHANIK-DRESDEN/ShapeOut#261]
We are working on a new release of Shape-In and in it, will allow the event index to be discontinuous in special (and rare) cases of data writer overrun leading to discarding of events for rescue.

Until now, event index was/is continuous even in these situations but for technical reasons and in spirit of best documentation we will change this.

Shape-Out seems to work with the event index from the 'index' feature in determining which image to show when clicking on a data point and when changing the image shown by the scroll box control. (And possibly Shape-Out and dclab uses it in different instances as identifier of a column in a data array?)

For continued correct functioning it would be necessary to change this behavior and use row# and image# from the .rtdc file instead.

Thanks,
Christoph

**Comment by paulmueller** [comment]
I strongly oppose this breaking change.

Not only will this make new data incompatible with older versions of dclab/Shape-Out, but it would also require refactoring in many places (think about indexing in hierarchy children) several of which I will probably not catch in the first round of refactoring. I will also have to write new test functions for this.

How about introducing a new feature ``index_online`` for this purpose? I would only have to add one line to dclab to support this.

**Comment by phidahl** [comment]
Ok, I understand your concerns.

I tested the following: the data field "index" was renamed to "index_online". ShapeOut 9.8.0 seems to load the data fine. Am I correctly understanding the function "compute_index" to create the index if it is not found in the data?
In this case by renaming the feature everything should be fine.

**Comment by paulmueller** [comment]
@phidahl Yes, if the "index" feature is not present, it will be automatically generated by this ancillary feature.
